src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.CSharpCodeGenerator.cs

```diff
@@ -667,42 +703,40 @@ protected StatementSyntax GetStatementContainingInvocationToExtractedMethodWorke
                 }
 
                 var syntaxNode = originalDocument.Root.GetAnnotatedNodesAndTokens(MethodDefinitionAnnotation).FirstOrDefault().AsNode();
-                if (syntaxNode == null || !(syntaxNode is MethodDeclarationSyntax methodDeclaration))
+                var nodeIsMethodOrLocalFunction = syntaxNode is MethodDeclarationSyntax || syntaxNode is LocalFunctionStatementSyntax;
+                if (syntaxNode == null || !nodeIsMethodOrLocalFunction)
                 {
                     return await base.UpdateMethodAfterGenerationAsync(originalDocument, methodSymbolResult, cancellationToken).ConfigureAwait(false);
                 }
 
                 var semanticModel = originalDocument.SemanticModel;
-                var methodOperation = semanticModel.GetOperation(methodDeclaration, cancellationToken);
 
-                var returnOperations = methodOperation.DescendantsAndSelf().OfType<IReturnOperation>();
-
-                foreach (var returnOperation in returnOperations)
+                if (syntaxNode is MethodDeclarationSyntax methodDeclaration)
                 {
-                    // If thereturn statement is located in a nested local function or lambda it
-                    // shouldn't contribute to the nullability of the extracted method's return type
-                    if (!ReturnOperationBelongsToMethod(returnOperation.Syntax, methodOperation.Syntax))
+                    var nullableReturnOperations = await CheckReturnOperations(methodDeclaration).ConfigureAwait(false);
+                    if (nullableReturnOperations is object)
                     {
-                        continue;
+                        return nullableReturnOperations;
                     }
 
-                    var syntax = returnOperation.ReturnedValue?.Syntax ?? returnOperation.Syntax;
-                    var returnTypeInfo = semanticModel.GetTypeInfo(syntax, cancellationToken);
-                    if (returnTypeInfo.Nullability.FlowState == NullableFlowState.MaybeNull)
+                    var newDocument = await GenerateNewDocument(methodDeclaration.ReturnType).ConfigureAwait(false);
+                    return await SemanticDocument.CreateAsync(newDocument, cancellationToken).ConfigureAwait(false);
+                }
+                else if (syntaxNode is LocalFunctionStatementSyntax localMethodDeclaration)
+                {
+                    var nullableReturnOperations = await CheckReturnOperations(localMethodDeclaration).ConfigureAwait(false);
+                    if (nullableReturnOperations != null)
```

Comment:
this logic is idnetical for both branches.  pull out and do before you switch on hte type of hte declaration syntax.